cmd/action/mux.go

```diff
@@ -90,14 +91,34 @@ func SchemaNameFromDSN(url string) (string, error) {
 		}
 		return cfg.DBName, err
 	case "postgres":
-		return "public", nil
+		return postgresSchema(dsn)
 	case "sqlite":
 		return schemaName(dsn)
 	default:
 		return "", fmt.Errorf("unknown database type: %q", key)
 	}
 }
 
+func postgresSchema(dsn string) (string, error) {
+	url, err := url.Parse(dsn)
+	if err != nil {
+		// For backwards compatibility, we default to "public" when failing to
+		// parse.
+		return "public", nil
+	}
+
+	// lib/pq supports setting default schemas via the `search_path` parameter
+	// in a dsn.
+	//
+	// See: https://github.com/lib/pq/blob/8446d16b8935fdf2b5c0fe333538ac395e3e1e4b/conn.go#L1155-L1165
+	queries := url.Query()
+	if queries.Has("search_path") {
+		return queries.Get("search_path"), nil
+	}
+
+	return "public", nil
```

Comment:
One comment, maybe skip it if the "search_path" is empty or return an error? (e.g. `"postgres://localhost:5432/dbname?search_path="`).

```suggestion
	url, err := url.Parse(dsn)
	if err != nil {
		// For backwards compatibility, we default to "public" when failing to
		// parse.
		return "public", nil
	}
	// lib/pq supports setting default schemas via the `search_path` parameter
	// in a dsn.
	//
	// See: https://github.com/lib/pq/blob/8446d16b8935fdf2b5c0fe333538ac395e3e1e4b/conn.go#L1155-L1165
	if schema := url.Query().Get("search_path"); schema != "" {
		return schema, nil
	}
```

Reply:
I approved the PR but then thought about the "empty value" case. 
cc @CGA1123 @masseelch